clojure-ts-mode.el

```diff
@@ -746,33 +746,61 @@ The possible values for this variable are
      ((parent-is "list_lit") parent 1)
      ((parent-is "set_lit") parent 2))))
 
-(defvar clojure-ts--symbols-with-body-expressions-regexp
+(defvar clojure-ts--symbols-block-0-regexp
   (eval-and-compile
-    (rx (or
-         ;; Match def* symbols,
-         ;; we also explicitly do not match symbols beginning with
-         ;; "default" "deflate" and "defer", like cljfmt
-         (and line-start "def")
-         ;; Match with-* symbols
-         (and line-start "with-")
-         ;; Exact matches
-         (and line-start
-              (or "alt!" "alt!!" "are" "as->"
-                  "binding" "bound-fn"
-                  "case" "catch" "comment" "cond" "condp" "cond->" "cond->>"
-                  "delay" "do" "doseq" "dotimes" "doto"
-                  "extend" "extend-protocol" "extend-type"
-                  "fdef" "finally" "fn" "for" "future"
-                  "go" "go-loop"
-                  "if" "if-let" "if-not" "if-some"
-                  "let" "letfn" "locking" "loop"
-                  "match" "ns" "proxy" "reify" "struct-map"
-                  "testing" "thread" "try"
-                  "use-fixtures"
-                  "when" "when-first" "when-let" "when-not" "when-some" "while")
-              line-end))))
+    (rx (and line-start
+             (or "alt!" "alt!!"
+                 "comment"
+                 "cond"
+                 "delay"
+                 "do"
+                 "finally"
+                 "future"
+                 "go"
+                 "thread"
+                 "try"
+                 "with-out-str")
+             line-end)))
   "A regex to match symbols that are functions/macros with a body argument.
-Taken from cljfmt:
+Taken from cljfmt (all symbols with [[:block 0]] rule):
+https://github.com/weavejester/cljfmt/blob/fb26b22f569724b05c93eb2502592dfc2de898c3/cljfmt/resources/cljfmt/indents/clojure.clj")
+
+(defun clojure-ts--match-block-0-body (node parent bol)
+  "Match NODE if it is an argument to a PARENT expression.
+
+Check if the first expression in the body is not at the same line as
+NODE.  If there is no body, check that BOL is not at the same line."
+  (and (clojure-ts--list-node-p parent)
+       (let* ((first-child (clojure-ts--node-child-skip-metadata parent 0))
+              (body-pos (if-let* ((body (treesit-node-next-sibling first-child)))
+                            (treesit-node-start body)
+                          bol)))
+         (and (not (clojure-ts--match-with-metadata node))
+              (clojure-ts--symbol-matches-p
+               clojure-ts--symbols-block-0-regexp
+               first-child)
+              (< (line-number-at-pos (treesit-node-start first-child))
+                 (line-number-at-pos body-pos))))))
+
+(defvar clojure-ts--symbols-block-1-regexp
+  (eval-and-compile
+    (rx (and line-start
+             (or "defprotocol"
```

Comment:
Basically, I'm suggesting to make lists like this one separate vars that users can modify, as they will likely have macros in their own code that need to be indented in a non-standard way.

Reply:
Done.